### PR TITLE
Teapot handler

### PR DESCRIPTION
**Pull request by td-extreme**
Created a `IHandler` for teapot and test that method `generateResponse()` returns an `HttpResponse` with its `responseCode` set to 418.

### CODE REVIEW

File: src/httpserver/handlers/teapot.clj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+(ns httpserver.handlers.teapot)
+
+(import '(com.td.HttpServer HttpResponse))
+(import '(com.td.HttpServer IHandler))
+
+(deftype teapotHandler []
+  com.td.HttpServer.IHandler
+  (generateResponse [this]
+    (def response (HttpResponse.))
+    (.setBody response "I'm a teapot", "text/plain")
```
Comment (dadarek) [inline src/httpserver/handlers/teapot.clj]: can you reference those constants here instead of `text/plain`?

**Comment by dadarek** [comment]
LGTM but then again I'm not a big clojure dude.

@mikeknep ?

File: src/httpserver/handlers/teapot.clj [hunk 2]
```diff
@@ -0,0 +1,15 @@
+(ns httpserver.handlers.teapot)
+
+(import '(com.td.HttpServer HttpResponse))
+(import '(com.td.HttpServer IHandler))
+
+(deftype teapotHandler []
+  com.td.HttpServer.IHandler
+  (generateResponse [this]
+    (def response (HttpResponse.))
```
Comment (mikeknep) [inline src/httpserver/handlers/teapot.clj]: You should use `let` here, similar to how you're using it in your specs

File: src/httpserver/handlers/teapot.clj (hunk 1 above)
Comment (td-extreme) [inline src/httpserver/handlers/teapot.clj]: Yes, I should be able to do that.

File: src/httpserver/handlers/teapot.clj [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    (.setBody response "I'm a teapot", "text/plain")
+    (.setResponseCode response 418)
+    response))
```
Comment (mikeknep) [inline src/httpserver/handlers/teapot.clj]: Some optional Clojure sugar:

``` clojure
(doto (HttpResponse.)
      (.setBody "I'm a teapot", "text/plain")
      (.setResponseCode 418))
```

**Comment by mikeknep** [comment]
The `let` is a crucial fix and a blocker. The `doto` refactor you can take or leave. Ship it after the `let` gets fixed

File: src/httpserver/handlers/teapot.clj (hunk 2 above)
Comment (td-extreme) [inline src/httpserver/handlers/teapot.clj]: Wilco

**Pull request merged**